pkg/playground/playground.go

```diff
@@ -44,19 +49,36 @@ type fileConfig struct {
 	fileContentType string
 }
 
-type HttpServeMux interface {
-	HandleFunc(pattern string, handler func(http.ResponseWriter, *http.Request))
+// HandlerConfig is the configuration Object for all playground http Handlers
+type HandlerConfig struct {
+	// URL is where the handler should be hosted
+	URL     string
+	// Handler is the http.HandlerFunc that should be hosted on the corresponding URL
+	Handler http.HandlerFunc
 }
 
-func ConfigureHandlers(mux HttpServeMux, config Config) error {
+// Handlers is an array of HandlerConfig
+// The playground expects that you make all assigned Handlers available on the corresponding URL
+type Handlers []HandlerConfig
+
+func (h *Handlers) add(url string, handler http.HandlerFunc) {
+	*h = append(*h, HandlerConfig{
+		URL:     url,
+		Handler: handler,
+	})
+}
+
+// ConfigureHandlers takes your Config and sets up all handlers to the supplied Handlers slice
+func ConfigureHandlers(config Config, handlers *Handlers) (err error) {
```

Comment:
Why do you think we have to supply handlers outside of this function?
Maybe replace this method with
```go
func NewPlaygroundHandlers(config Config) (handlers *Handlers, err error) {
}
```

Reply:
another possible way
```go
type Playground struct {
  cfg Config
  box packr.Box
  files
}

func NewPlayground(config Config) (*Playground, err error) {
   return &Playground{
     cfg: config,
     box: packr.NewBox("./files"),
     files: ...
   }
}
func (h *Plaground) GetHandlers() (Handlers, error) {
}
func (h *Plaground) addFileHandler(fileName, fileURL, contentType string) error {
}

```